lib/segment/src/index/field_index/mmap_point_to_values.rs

```diff
@@ -0,0 +1,389 @@
+        let header =
+            Header::read_from_prefix(mmap.as_ref()).ok_or(OperationError::InconsistentStorage {
+                description: NOT_ENOUGHT_BYTES_ERROR_MESSAGE.to_owned(),
+            })?;
+        let mmap = unsafe { MmapSlice::try_from(mmap)? };
+
+        Ok(Self {
+            file_name,
+            mmap,
+            header,
+        })
+    }
+
+    fn flusher(&self) -> Flusher {
+        self.mmap.flusher()
+    }
+
+    fn files(&self) -> Vec<PathBuf> {
+        vec![self.file_name.clone()]
+    }
+
+    pub fn get_values<'a, T: MmapValue<'a>>(
+        &'a self,
+        point_id: PointOffsetType,
+    ) -> OperationResult<Box<dyn Iterator<Item = OperationResult<T>> + 'a>> {
+        // first, get range of values for point
+        let range = if point_id < self.header.points_count as PointOffsetType {
+            let range_offset = (self.header.ranges_start as usize)
+                + (point_id as usize) * std::mem::size_of::<MmapRange>();
+            self.mmap
+                .get(range_offset..range_offset + std::mem::size_of::<MmapRange>())
+                .and_then(MmapRange::read_from)
+                .unwrap_or_default()
+        } else {
+            MmapRange { start: 0, count: 0 }
+        };
+
+        // second, define iteration step for values
+        // iteration step gets remainder range from memmaped file and returns next range with left range
+        let bytes: &[u8] = self.mmap.as_ref();
+        let read_value = move |range: MmapRange| -> Option<(OperationResult<T>, MmapRange)> {
+            if range.count > 0 {
+                let value = T::read_from_mmap(bytes, range.start as usize);
+                match value {
+                    Ok(value) => {
+                        let range = MmapRange {
+                            start: range.start + value.mmaped_size() as u64,
+                            count: range.count - 1,
+                        };
+                        Some((Ok(value), range))
+                    }
+                    // if error occurs, return error and empty range to stop iteration
+                    Err(err) => Some((Err(err), MmapRange::default())),
+                }
+            } else {
+                None
+            }
+        };
+
+        // finally, return iterator
+        Ok(Box::new(
+            std::iter::successors(read_value(range), move |range| read_value(range.1))
+                .map(|(value, _)| value),
+        ))
+    }
+
+    pub fn remove_point(&mut self, point_id: PointOffsetType) -> OperationResult<()> {
+        // change points values range to empty
+        let range_offset = (self.header.ranges_start as usize)
+            + (point_id as usize) * std::mem::size_of::<MmapRange>();
+        if let Some(range) = self
+            .mmap
+            .get_mut(range_offset..range_offset + std::mem::size_of::<MmapRange>())
+            .and_then(MmapRange::mut_from)
+        {
+            *range = Default::default();
```

Comment:
Does this just zero everything in the range of an element?

If so I'd like to have it a bit more explicit.

We can do [`zeroize()`](https://docs.rs/zeroize/latest/zeroize/trait.Zeroize.html#tymethod.zeroize) since that crate is already in our tree.

But a comment would also suffice.

Reply:
Thanks! Fixed. I use an empty iterator instead as the most explicit way to handle such points

Reply:
After discussion, we decided to use deleted flags on a higher level and this map does not support deletion anymore. Deleted the whole function